### PR TITLE
[CARBONDATA-328] [Spark] Improve Code and Fix Warnings [Squashed]

### PR DESCRIPTION
**Pull request by PKOfficial**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[CARBONDATA-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).
 - [x] Testing done
- [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
 
**Please provide details on**
- **Whether new unit test cases have been added or why no new tests are required?**
Not required because no change in functionality.
- **What manual testing you have done?**
Run basic Commands in Beeline.
- **Any additional information to help reviewers in testing this change.**
No
                 

---
Improved spark module code.
    * Removed some compliation warnings.
    * Replace pattern matching for boolean to IF-ELSE.
    * Improved code according to scala standards.
    * Removed unnecessary new lines.
    * Added string interpolation instead of string concatenation.
    * Removed unnecessary semi-colons.
    * Fixed indentation.
    * add useKettle option for loading
    * Fixed indentation.

### CODE REVIEW

**Comment by PKOfficial** [comment]
@jackylk I've rebased and squashed it into one commit.

**Comment by jackylk** [comment]
please rebase

**Comment by PKOfficial** [comment]
@jackylk Done with the rebase. Please have a look.

File: integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala [hunk 1]
```diff
@@ -32,7 +32,10 @@ import org.apache.hadoop.mapreduce.Job
 import org.apache.hadoop.mapreduce.lib.input.{FileInputFormat, FileSplit}
 import org.apache.spark.{util => _, _}
 import org.apache.spark.sql.{CarbonEnv, DataFrame, SQLContext}
-import org.apache.spark.sql.execution.command.{AlterTableModel, CompactionCallableModel, CompactionModel, Partitioner}
+
```
Comment (jackylk) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: code style check failed, remove this line

File: integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala [hunk 2]
```diff
@@ -32,7 +32,10 @@ import org.apache.hadoop.mapreduce.Job
...
+import org.apache.spark.sql.execution.command.{AlterTableModel, CompactionCallableModel,
+CompactionModel, Partitioner}
+
```
Comment (jackylk) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: code style check failed, remove this line

**Comment by jackylk** [comment]
@PKOfficial remove these two lines, and it can be merged

**Comment by PKOfficial** [comment]
@jackylk Removed those line.

**Pull request closed without merging**